pkg/e2e/e2e_test.go

```diff
@@ -64,31 +66,33 @@ func TestMain(m *testing.M) {
 		klog.Fatal("kind default config set node count to 0")
 	}
 
-	kubeadmConfig := `metadata:
+	if v.Compare(v13) < 0 {
+		kubeadmConfig := `metadata:
   name: config
 networking:
   serviceSubnet: 10.0.0.0/16`
 
-	if v.Compare(v13) >= 0 {
-		kubeadmConfig = fmt.Sprint(`apiVersion: kubeadm.k8s.io/v1beta1
-kind: ClusterConfiguration
-`, kubeadmConfig)
-	} else if v.Compare(v12) < 0 {
-		kubeadmConfig = fmt.Sprint(`apiVersion: kubeadm.k8s.io/v1alpha2
+		if v.Compare(v12) < 0 {
+			kubeadmConfig = fmt.Sprint(`apiVersion: kubeadm.k8s.io/v1alpha2
 kind: MasterConfiguration
 `, kubeadmConfig)
-	} else {
-		kubeadmConfig = fmt.Sprint(`apiVersion: kubeadm.k8s.io/v1alpha3
+		} else {
+			kubeadmConfig = fmt.Sprint(`apiVersion: kubeadm.k8s.io/v1alpha3
 kind: ClusterConfiguration
 `, kubeadmConfig)
-	}
+		}
 
-	conf.KubeadmConfigPatches = []string{kubeadmConfig}
+		conf.KubeadmConfigPatches = []string{kubeadmConfig}
+	} else {
+		conf.Networking.ServiceSubnet = "10.0.0.0/16"
```

Comment:
Not sure if this is new code or not, but how come you need to set up a custom service CIDR?

Reply:
Was causing circular DNS issues in CI